src/manifest.rs

```diff
@@ -65,262 +65,153 @@ impl From<bare_version::Error> for ManifestParseError {
+        metadata
+            .root_package()?
+            .metadata
+            .as_object()?
+            .get("msrv")?
+            .as_str()
     }
 
     // Parse the MSRV from the `package.rust-version` key if it exists,
     // and try to fallback to our own `package.metadata.msrv` if it doesn't
-    find_rust_version(document).or_else(|| find_metadata_msrv(document))
+    match find_rust_version(metadata) {
+        Some(version) => Ok(Some(BareVersion::from(version))),
+        None => find_metadata_msrv(metadata)
+            .map(BareVersion::try_from)
+            .transpose(),
+    }
 }
 
 #[cfg(test)]
 mod minimal_version_tests {
-    use crate::manifest::bare_version::Error;
-    use crate::manifest::{
-        BareVersion, CargoManifest, CargoManifestParser, ManifestParseError, TomlParser,
-    };
+    use crate::manifest::{BareVersion, CargoManifest};
+    use cargo_metadata::Metadata;
     use std::convert::TryFrom;
-    use toml_edit::Document;
 
-    #[test]
-    fn parse_toml() {
-        let contents = r#"[package]
-name = "some"
-version = "0.1.0"
-edition = "2018"
-
-[dependencies]
-"#;
-
-        assert!(CargoManifestParser.parse::<Document>(contents).is_ok());
-    }
-
-    #[test]
-    fn parse_invalid_toml() {
-        let contents = r#"-[package]
-name = "some"
-version = "0.1.0"
-edition = "2018"
-
-[dependencies]
-"#;
-
-        assert!(CargoManifestParser.parse::<Document>(contents).is_err());
+    fn metadata_json(rust_version: Option<&str>, metadata: Option<&str>) -> String {
+        let rust_version = match rust_version {
+            Some(rust_version) => format!(r#""rust_version": "{}","#, rust_version),
+            None => "".to_string(),
+        };
+        let metadata = match metadata {
+            Some(metadata) => format!(r#""metadata": {},"#, metadata),
+            None => "".to_string(),
+        };
+        format!(
+            r#"{{
+  "packages": [
+    {{
+      "name": "some",
+      "version": "0.1.0",
+      "id": "some 0.1.0 (path+file:///some)",
+      "manifest_path": "/some/Cargo.toml",
+      {}
+      {}
+      "dependencies": [],
+      "targets": [],
+      "features": {{}},
+      "edition": "2018"
+    }}
+  ],
+  "workspace_members": [
+    "some 0.1.0 (path+file:///some)"
+  ],
+  "target_directory": "/some/target",
+  "version": 1,
+  "workspace_root": "/some"
+}}"#,
+            rust_version, metadata
+        )
     }
 
     #[test]
     fn parse_no_minimum_rust_version() {
-        let contents = r#"[package]
-name = "some"
-version = "0.1.0"
-edition = "2018"
+        let metadata = metadata_json(None, None);
+        let metadata: Metadata = serde_json::from_str(&metadata).unwrap();
 
-[dependencies]
-"#;
-
-        let manifest = CargoManifestParser.parse::<Document>(contents).unwrap();
-
-        let manifest = CargoManifest::try_from(manifest).unwrap();
+        let manifest = CargoManifest::try_from(metadata).unwrap();
 
```

Comment:
Figured it was easier to test by passing in the JSON from cargo_metadata, instead of trying to call cargo metadata on a TOML (which would require tempfile or adding more Cargo.toml test case files to the repo)

As such, also removed tests that were about parsing TOML only, since cargo metadata will handle that now